### ISSUE TITLE
Documentation: Building PDF version of docs results in error

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3134]
l.3553 \begin{itemize}

@pdurbin I think you said you were familiar with this problem? Can you help?

!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on Dataverse.log.
make[1]: **\* [Dataverse.pdf] Error 1
make[1]: Leaving directory `/var/lib/jenkins/workspace/guides.dataverse.org-rfi/doc/sphinx-guides/build/latex'
make: **\* [latexpdf] Error 2
Build step 'Execute shell' marked build as failure
Finished: FAILURE

**Comment by pdurbin** [comment]
Here's more of the error as of a build from c81b7cf9e1c5486361bfd0d7b720eef1f231dbca

```
! LaTeX Error: Too deeply nested.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.3553 \begin{itemize}
                      
? 
! Emergency stop.
 ...                                              
                                                  
l.3553 \begin{itemize}
                      
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on Dataverse.log.
make[1]: *** [Dataverse.pdf] Error 1
make[1]: Leaving directory `/var/lib/jenkins/workspace/guides.dataverse.org-rfi/doc/sphinx-guides/build/latex'
make: *** [latexpdf] Error 2
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

The last successful build was from 11ad621673d569a85287c0d12da1e95a2b89a0d9 and ended with this:

```
Chapter 7.
(./Dataverse.ind) [117] (./Dataverse.aux) )
(see the transcript file for additional information){/nfs/tools/apps/texlive/20
15/texmf-dist/fonts/enc/dvips/base/8r.enc}</nfs/tools/apps/texlive/2015/texmf-d
ist/fonts/type1/urw/courier/ucrb8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist
/fonts/type1/urw/courier/ucrr8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/fo
nts/type1/urw/courier/ucrro8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/font
s/type1/urw/helvetic/uhvb8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/fonts/
type1/urw/helvetic/uhvbo8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/fonts/t
ype1/urw/times/utmb8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/fonts/type1/
urw/times/utmr8a.pfb></nfs/tools/apps/texlive/2015/texmf-dist/fonts/type1/urw/t
imes/utmri8a.pfb>
Output written on Dataverse.pdf (121 pages, 1162853 bytes).
Transcript written on Dataverse.log.
make[1]: Leaving directory `/var/lib/jenkins/workspace/guides.dataverse.org-rfi/doc/sphinx-guides/build/latex'
pdflatex finished; the PDF files are in build/latex.
+ cp build/latex/Dataverse.pdf build/html
```

So I guess something bad happened between 11ad621 (which shipped as part of 4.3.1) and c81b7cf (which is in the 2469-widgets branch). The 2469-widgets branch is pull request #3118 and the only change to the Sphinx source files so far was 873e341 so I'm guessing the problem was introduced there but I don't really know. As a workaround @kcondon and I decided to stop trying to build the PDF version since we don't even link to them yet per #2148.

@scolapasta I'm passing this to you for review.

**Comment by scolapasta** [comment]
Reverted the doc change from he widgets branch and it now works.

(we still need to work on the doc for widgets in general, so we'll make sure this continues to work when that is complete.

**Comment by mheppler** [comment]
Created a new issue to track the work needed for the widget documentation. #3147

**Comment by kcondon** [comment]
The error still happens with the most recent build. Passing back.

**Comment by scolapasta** [comment]
@pdurbin can you take a look with @mheppler and @eaquigley who have made doc changes and figure out what is causing this to break? Thanks.

**Comment by pdurbin** [comment]
@eaquigley is taking a look so I'll assign this to her. Something about nesting.

**Comment by kcondon** [comment]
There are also numerous latex build warnings leading up to the Too Deeply Nested error and most of those have to do with the Installation guide, shibboleth section but others too:

LaTeX Warning: Hyper reference `installation/shibboleth:shibboleth2-xml' on pag
e 63 undefined on input line 3595.


LaTeX Warning: Hyper reference `installation/shibboleth:specific-identity-provi
der-s-vs-identity-federation' on page 63 undefined on input line 3598.


! LaTeX Error: Too deeply nested.

**Comment by pdurbin** [comment]
Huh. If nesting in the Shib docs are a problem I do plan to revisit those docs as part of #2937.

The real problem here is that developers don't know how to create the PDF version of the guides on their laptops. I just tried this on my MacBook and wasn't able to figure out out quickly. If we ever make PDF versions of the guides a first class citizen (#2148) it will be considered a regression if a developer breaks the PDF. Developers will need to test locally before committing changes.

**Comment by kcondon** [comment]
Right we just talked about this: we should be looking at warnings in the doc builds and understanding how to avoid them but the specific ones mentioned can likely be deferred for this release until we have time to revisit or, worst case, when we implement printable versions using pdf.

**Comment by pdurbin** [comment]
Did we ever successfully build a pdf for 4.3 or 4.3.1? The latest I could find was http://guides.dataverse.org/en/4.2.4/Dataverse.pdf so maybe PDF generation has been broken for a while.

**Comment by kcondon** [comment]
We only recently began seeing guide build errors which prompted this ticket.

**Comment by pdurbin** [comment]
As I mentioned to @kcondon yesterday I spend maybe 15 minutes trying to get a PDF to build in Vagrant as of 75031ba but with a random version of Sphinx (1.4.4). I'm getting this error...

```
(/usr/share/texmf/tex/latex/config/hyperref.cfg)
(/usr/share/texmf/tex/latex/oberdiek/kvoptions.sty)

! Package keyval Error: pdfencoding undefined.

See the keyval package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.2238 \ProcessKeyvalOptions{Hyp}
                                 
? 
```

... then I hit return and I see this...

```
LaTeX Warning: Hyper reference `installation/shibboleth:specific-identity-provi
der-s-vs-identity-federation' on page 66 undefined on input line 3616.


! LaTeX Error: Too deeply nested.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.3617 \begin{itemize}
                      
? 
```

... I hit return again and a PDF is produced but I don't know why I have to hit return. On Jenkins we use Sphinx 1.1.3. I don't feel like the dependencies and versions for building the PDFs are captured or even well understood. We should probably have https://readthedocs.org build the docs for us at some point.

Anyway, I expressed to @djbrooke that this whole issue of not having a PDF version of the guides feels like a duplicate of #2148 to me and he gave his blessing to move this issue out of 4.4, which I'll do now.

**Comment by mheppler** [comment]
Fixed nesting format issue in Shibboleth > Installation Guide breaking PDF creation of the guides. Also fixed other formatting, broken section link ref's and download file ref warnings.

```
Output written on Dataverse.pdf (147 pages, 1541202 bytes).
Transcript written on Dataverse.log.
pdflatex finished; the PDF files are in build/latex.
```

**Comment by pdurbin** [comment]
@mheppler thanks! I added "Make it so PDFs can be built again (#3134). Start by merging 3134-docs-build-errors into 3338-oauth-login" to the checklist in #3338.

**Comment by pdurbin** [comment]
The fixes by @mheppler looked good. I only had to make one small change at 843e612.

In order to run `make latexpdf` I had to run `brew cask install mactex`. We should probably document this at http://guides.dataverse.org/en/4.6/developers/documentation.html so developers know how to even build a PDF on their Mac.

I just merged the 3134-docs-build-errors branch into into 3338-oauth-login so I'm going to delete the former. The latter should be used for QA and I'm moving this card there in https://waffle.io/IQSS/dataverse
